src/__tests__/test-utils.test.tsx

```diff
@@ -0,0 +1,174 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { ComponentType } from "react";
+import { render } from "@testing-library/react";
+import { paramCase, pascalCase } from "change-case";
+import { describe, expect, test } from "vitest";
+
+import * as components from "../../lib/components";
+import createWrapperDom, { ElementWrapper as DomElementWrapper } from "../../lib/components/test-utils/dom";
+import createWrapperSelectors from "../../lib/components/test-utils/selectors";
+import { ItemContextWrapper } from "../board-item/__tests__/board-item-wrapper";
+import { defaultProps } from "./default-props";
+
+const componentWithMultipleRootElements = ["Board", "ItemsPalette"];
+const componentNames = Object.keys(components).filter(
+  (component) => !componentWithMultipleRootElements.includes(component),
+);
+
+const RENDER_COMPONENTS_DEFAULT_PROPS: Record<string, unknown>[] = [
+  {
+    "data-testid": "first-item",
+    "data-name": "first item",
+  },
+  {
+    "data-testid": "second-item",
+    "data-name": "second item",
+  },
+];
+
+function renderComponents(componentName: string, props = RENDER_COMPONENTS_DEFAULT_PROPS) {
+  const Component = components[componentName as keyof typeof components] as ComponentType;
+  const componentDefaultProps = defaultProps[paramCase(componentName) as keyof typeof defaultProps];
+  return render(
+    <div>
+      {props.map(({ ...customProps }, index) => (
+        <Component key={index} {...componentDefaultProps} {...customProps} />
+      ))}
+    </div>,
+    { wrapper: ItemContextWrapper },
+  );
+}
+
+function getComponentSelectors(componentName: string) {
+  const componentNamePascalCase = pascalCase(componentName);
+  const findAllRegex = new RegExp(`findAll${componentNamePascalCase}.*`);
```

Comment:
There is a lot of transformations happening for a test file. Wouldn't it be better straightforward to have some mapping instead, like:

```
const components = [
  {
    name: 'Board',
    component: components.Board,
    findName: 'findBoard',
    findAllName: 'findAllBoard',
  },
]
```

Reply:
Whatever practice we go for, we need to make it consistent all across the repos. So gotta have the map also in the components repo and three other repos as well.

I agree with the concern, but I think we need to fix this a bit differently and outside of the scope of this PR.